In [1]:
import pandas as pd
import numpy as np

## OFSP original and OFSP English were given by Dr Thai Son Mai. OFSP_rpt_rmv was created in Excel by Amonae Dabbs-Brown

these are their stories

In [2]:
OFSP_original = pd.read_csv("OSFP_2019_Original.csv", encoding= 'unicode_escape') # original unaltered df
OFSP_original.head()

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3049: DtypeWarning: Columns (0,2,3,4,5,6,7,8,9,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,42,43,45,46,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,170,171,172,173,174,175,176,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,211,213,214,215,216,217,218,219,220,221,222,223,224,225,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264

 Patient ID Date of birth Department’s id Country’s ID        Sex  \
0   id_patient           ddn  id_departement_P      id_pays_P  id_sexe_P   
1            1     21-Nov-73                33             74          2   
2            1           NaN               NaN            NaN        NaN   
3            1           NaN               NaN            NaN        NaN   
4            1           NaN               NaN            NaN        NaN   

        Profession     Current smoker Number of current cigarettes package  \
0  id_profession_P  tabagismeActuel_P                    nbPaquetsActuel_P   
1                5                  0                                  NaN   
2              NaN                NaN                                  NaN   
3              NaN                NaN                                  NaN   
4              NaN                NaN                                  NaN   

      Former smoker  Number of former cigarettes package  ...  \
0  tabagismeAncien_P                   nbPaquetsAncien_P  ...   
1                  1                                   5  ...   
2                NaN                                 NaN  ...   
3                NaN                                 NaN  ...   
4                NaN                                 NaN  ...   

  Treatment name.38 Treatment name.39 Short treatment name .18   ATC code.15  \
0          medoc_20         medoc2_20             nom_court_20    code_atc_20   
1               NaN               NaN                      NaN            NaN   
2               NaN               NaN                      NaN            NaN   
3               NaN               NaN                      NaN            NaN   
4               NaN               NaN                      NaN            NaN   

   ATC class .19            Dosage.32        Dosage.33 Table :  \
0  classe_atc_20  posologie_valeur_20  id_posologie_20    table   
1            NaN                  NaN              NaN        1   
2            NaN                  NaN              NaN        2   
3            NaN                  NaN              NaN        2   
4            NaN                  NaN              NaN        2   

  Number of the visist     Incorrect visit  
0           num_visite  visite1_incoherent  
1                    1                 NaN  
2                    2                 NaN  
3                    3                 NaN  
4                    4                 NaN  

[5 rows x 390 columns]

In [3]:
OFSP_original.shape

(190538, 390)

In [4]:
OFSP_English = pd.read_csv("OFSP-English.csv", encoding= 'unicode_escape') # original df with labels translated to englsih
OFSP_English.head()

Patient ID Date of birth   Department’s id Country’s ID        Sex  \
0  id_patient           ddn  id_departement_P    id_pays_P  id_sexe_P   
1           1     21-Nov-73                33           74          2   
2           1           NaN               NaN          NaN        NaN   
3           1           NaN               NaN          NaN        NaN   
4           1           NaN               NaN          NaN        NaN   

        Profession     Current smoker Number of current cigarettes package  \
0  id_profession_P  tabagismeActuel_P                    nbPaquetsActuel_P   
1                5                  0                                  NaN   
2              NaN                NaN                                  NaN   
3              NaN                NaN                                  NaN   
4              NaN                NaN                                  NaN   

      Former smoker  Number of former cigarettes package  ...  \
0  tabagismeAncien_P                   nbPaquetsAncien_P  ...   
1                  1                                   5  ...   
2                NaN                                 NaN  ...   
3                NaN                                 NaN  ...   
4                NaN                                 NaN  ...   

  Treatment name.38 Treatment name.39 Short treatment name .18   ATC code.15  \
0          medoc_20         medoc2_20             nom_court_20   code_atc_20   
1               NaN               NaN                      NaN           NaN   
2               NaN               NaN                      NaN           NaN   
3               NaN               NaN                      NaN           NaN   
4               NaN               NaN                      NaN           NaN   

   ATC class .19            Dosage.32        Dosage.33 Table :  \
0  classe_atc_20  posologie_valeur_20  id_posologie_20   table   
1            NaN                  NaN              NaN       1   
2            NaN                  NaN              NaN       2   
3            NaN                  NaN              NaN       2   
4            NaN                  NaN              NaN       2   

  Number of the visist     Incorrect visit  
0           num_visite  visite1_incoherent  
1                    1                 NaN  
2                    2                 NaN  
3                    3                 NaN  
4                    4                 NaN  

[5 rows x 390 columns]

In [5]:
OFSP_English.shape

(190538, 390)

In [6]:
OFSP_English_rpts_rmv = pd.read_csv("OFSP_English_rpts_rmv.csv", encoding= 'unicode_escape') # OFSP_english with repeated columns removed and 1st row removed
OFSP_English_rpts_rmv.head()

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3049: DtypeWarning: Columns (2,29,149,150,155,190,195,196,197,198,199,203,204,205,206,207,211,212,213,214,215,219,220,221,222,223,227,228,229,230,231,235,236,237,238,239,243,244,245,246,247,251,252,253,254,255,259,260,261,262,263,267,268,269,270,271,275,276,277,278,279,283,284,285,286,287,291,292,293,294,295,299,301,302,303,307,309,310,311,315,316,317,318,319,323,325,326,327,334,339,342) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


PatientID Dateofbirth Departmentid  CountryID  Sex  Profession  \
0          1   21-Nov-73           33       74.0  2.0         5.0   
1          1         NaN          NaN        NaN  NaN         NaN   
2          1         NaN          NaN        NaN  NaN         NaN   
3          1         NaN          NaN        NaN  NaN         NaN   
4          2   10-Nov-33           17       74.0  1.0         7.0   

          Smoker   ____  ____.1   UserID  ...  Treatmentname.38  \
0  fumeur ancien    0.0     NaN    543.0  ...               NaN   
1  fumeur ancien    NaN     NaN      NaN  ...               NaN   
2  fumeur ancien    NaN     NaN      NaN  ...               NaN   
3  fumeur ancien    NaN     NaN      NaN  ...               NaN   
4     non fumeur    0.0     NaN    543.0  ...               NaN   

  Treatmentname.39  Shorttreatmentname.18   ATCcode.15 ATCclass.19  Dosage.32  \
0              NaN                    NaN          NaN         NaN        NaN   
1              NaN                    NaN          NaN         NaN        NaN   
2              NaN                    NaN          NaN         NaN        NaN   
3              NaN                    NaN          NaN         NaN        NaN   
4              NaN                    NaN          NaN         NaN        NaN   

   Dosage.33  Table :  Numberofthevisist  Incorrectvisit  
0        NaN        1                  1             NaN  
1        NaN        2                  2             NaN  
2        NaN        2                  3             NaN  
3        NaN        2                  4             NaN  
4        NaN        1                  1             NaN  

[5 rows x 357 columns]

In [7]:
OFSP_English_rpts_rmv.shape

(190537, 357)

## Comparing OFSP_original with OFSP_English

OFSP_original is the original file in French while OFSP English is the English translation of the same file. There should be no differences in cell contents

In [8]:
OFSP_original.equals(OFSP_English) # Check to see if the files are exactly alike. They are not

False

In [9]:
for i in range(390):
    if OFSP_original.iloc[:,i].equals(OFSP_English.iloc[:,i]):
        continue
    print(i)
# this shows that 11 columns have different data

136
177
182
207
209
212
230
238
246
254
262


In [10]:
different_cols = [136,177,182, 207, 209, 212, 230, 238, 246, 254, 262]
for col in different_cols:
    print(OFSP_English.columns[col])

# These are the columns with differenes

Average duration sleep estimate
 Sleep examination date
Saturation cumulated time 
CPAP observance 
Mean CPAP observance
Fan name
 Short treatment name 
Short treatment name 
Short treatment name .1
Short treatment name .2
Short treatment name .3


### Exploring differences

In [11]:
diff_o_e = {}
    
for col in different_cols:
    diff_o_e["diff{0}".format(col)] = pd.concat([OFSP_original.iloc[:,col], OFSP_English.iloc[:,col]], axis = 1)
# this creates a dictionary of set differences by column index 

In [12]:
diff_o_e.keys()

dict_keys(['diff136', 'diff177', 'diff182', 'diff207', 'diff209', 'diff212', 'diff230', 'diff238', 'diff246', 'diff254', 'diff262'])

In [13]:
for key in diff_o_e.keys():
    print(key, 'length', len(diff_o_e[key]))
    
# Each key is a df with 2 columns, one containing column data from OFSP_original and the other from OFSP_English and all rows

diff136 length 190538
diff177 length 190538
diff182 length 190538
diff207 length 190538
diff209 length 190538
diff212 length 190538
diff230 length 190538
diff238 length 190538
diff246 length 190538
diff254 length 190538
diff262 length 190538


In [14]:
globals().update(diff_o_e)
for key in diff_o_e.keys():
    x = globals()[key]
    diff_o_e[key] = x[x.iloc[:,0] != x.iloc[:,1]].dropna()
    
# This iterates through the keys (dfs) and drops duplicates where the value at OFSP_original is the same as OFSP English
# It also drops NAs
#Only rows where the 2 dfs have different values are retained

In [15]:
for key in diff_o_e.keys():
    print(key, 'length', len(diff_o_e[key]))
 # lengths of the dfs

diff136 length 155
diff177 length 91017
diff182 length 1
diff207 length 23
diff209 length 18
diff212 length 198
diff230 length 4
diff238 length 2
diff246 length 2
diff254 length 7
diff262 length 1


In [16]:
globals().update(diff_o_e) # converts keys into variables

### Differences between DFs

In [17]:
diff136.head()

Average duration sleep estimate Average duration sleep estimate
23123                         AOÛT-45                         AOÛT-45
76484                             è                              è
97745                          7à8H                           7à8H
107942                            è                              è
112237                          6à7                            6à7

In [18]:
diff177.head()

 Sleep examination date  Sleep examination date
1                19-02-2009              19/02/2009
7                17-02-2009              17/02/2009
8                17-02-2009              17/02/2009
21               10-02-2009              10/02/2009
22               10-02-2009              10/02/2009

In [19]:
diff182.head()

Saturation cumulated time  Saturation cumulated time 
175830                      4ù                        4ù

In [20]:
diff207.head() 

CPAP observance  CPAP observance 
31353         AOÛT-43          AOÛT-43
37049         AOÛT-30          AOÛT-30
48734         AOÛT-20          AOÛT-20
51789         AOÛT-15          AOÛT-15
64379         AOÛT-50          AOÛT-50

In [21]:
diff209.head() 

Mean CPAP observance Mean CPAP observance
17247             AOÛT-30              AOÛT-30
37049             AOÛT-30              AOÛT-30
48734             AOÛT-20              AOÛT-20
69847             07:"é               07:"é
74332             AOÛT-43              AOÛT-43

In [22]:
diff212.head() 

Fan name          Fan name
820  420E (6-9cmH²O)  420E (6-9cmH²O)
821  420E (6-9cmH²O)  420E (6-9cmH²O)
894          morphée          morphée
895          morphée          morphée
896          morphée          morphée

In [23]:
diff230.head() 

 Short treatment name   \
29450               COLOPRIV 200MG GELULE Gé 30   
44585               COLOPRIV 200MG GELULE Gé 30   
44586               COLOPRIV 200MG GELULE Gé 30   
150480  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   

                          Short treatment name   
29450               COLOPRIV 200MG GELULE Gé 30  
44585               COLOPRIV 200MG GELULE Gé 30  
44586               COLOPRIV 200MG GELULE Gé 30  
150480  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14

In [24]:
diff238.head() 

Short treatment name        Short treatment name 
25030  GLUCOLESS 850 MG Gé CPR 30  GLUCOLESS 850 MG Gé CPR 30
25031  GLUCOLESS 850 MG Gé CPR 30  GLUCOLESS 850 MG Gé CPR 30

In [25]:
diff246.head() 

Short treatment name .1  \
162479  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   
181766  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   

                        Short treatment name .1  
162479  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14  
181766  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14

In [26]:
diff254.head()

Short treatment name .2  \
120676  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   
120677  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   
120702  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   
120703  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   
120704  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   

                        Short treatment name .2  
120676  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14  
120677  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14  
120702  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14  
120703  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14  
120704  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14

In [27]:
diff262.head() 

Short treatment name .3  \
166540  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14   

                        Short treatment name .3  
166540  EMAPANT Gé 40 MG CPR GASTRORESISTANT 14

#### diff177  is the largest set diff group and it is because some dates are written "02-02-2022" and some are written "02/02/2022"
Moving on because none of these columns are included in analysis

## Comparing  OFSP_English with OFSP_English_rpts_rmv
 
 rpts_rmv refers to the fact that in  this file, repeated column names from OFSP_English are removed

In [28]:
OFSP_English.head()

Patient ID Date of birth   Department’s id Country’s ID        Sex  \
0  id_patient           ddn  id_departement_P    id_pays_P  id_sexe_P   
1           1     21-Nov-73                33           74          2   
2           1           NaN               NaN          NaN        NaN   
3           1           NaN               NaN          NaN        NaN   
4           1           NaN               NaN          NaN        NaN   

        Profession     Current smoker Number of current cigarettes package  \
0  id_profession_P  tabagismeActuel_P                    nbPaquetsActuel_P   
1                5                  0                                  NaN   
2              NaN                NaN                                  NaN   
3              NaN                NaN                                  NaN   
4              NaN                NaN                                  NaN   

      Former smoker  Number of former cigarettes package  ...  \
0  tabagismeAncien_P                   nbPaquetsAncien_P  ...   
1                  1                                   5  ...   
2                NaN                                 NaN  ...   
3                NaN                                 NaN  ...   
4                NaN                                 NaN  ...   

  Treatment name.38 Treatment name.39 Short treatment name .18   ATC code.15  \
0          medoc_20         medoc2_20             nom_court_20   code_atc_20   
1               NaN               NaN                      NaN           NaN   
2               NaN               NaN                      NaN           NaN   
3               NaN               NaN                      NaN           NaN   
4               NaN               NaN                      NaN           NaN   

   ATC class .19            Dosage.32        Dosage.33 Table :  \
0  classe_atc_20  posologie_valeur_20  id_posologie_20   table   
1            NaN                  NaN              NaN       1   
2            NaN                  NaN              NaN       2   
3            NaN                  NaN              NaN       2   
4            NaN                  NaN              NaN       2   

  Number of the visist     Incorrect visit  
0           num_visite  visite1_incoherent  
1                    1                 NaN  
2                    2                 NaN  
3                    3                 NaN  
4                    4                 NaN  

[5 rows x 390 columns]

In [29]:
OFSP_English_rpts_rmv.head()

PatientID Dateofbirth Departmentid  CountryID  Sex  Profession  \
0          1   21-Nov-73           33       74.0  2.0         5.0   
1          1         NaN          NaN        NaN  NaN         NaN   
2          1         NaN          NaN        NaN  NaN         NaN   
3          1         NaN          NaN        NaN  NaN         NaN   
4          2   10-Nov-33           17       74.0  1.0         7.0   

          Smoker   ____  ____.1   UserID  ...  Treatmentname.38  \
0  fumeur ancien    0.0     NaN    543.0  ...               NaN   
1  fumeur ancien    NaN     NaN      NaN  ...               NaN   
2  fumeur ancien    NaN     NaN      NaN  ...               NaN   
3  fumeur ancien    NaN     NaN      NaN  ...               NaN   
4     non fumeur    0.0     NaN    543.0  ...               NaN   

  Treatmentname.39  Shorttreatmentname.18   ATCcode.15 ATCclass.19  Dosage.32  \
0              NaN                    NaN          NaN         NaN        NaN   
1              NaN                    NaN          NaN         NaN        NaN   
2              NaN                    NaN          NaN         NaN        NaN   
3              NaN                    NaN          NaN         NaN        NaN   
4              NaN                    NaN          NaN         NaN        NaN   

   Dosage.33  Table :  Numberofthevisist  Incorrectvisit  
0        NaN        1                  1             NaN  
1        NaN        2                  2             NaN  
2        NaN        2                  3             NaN  
3        NaN        2                  4             NaN  
4        NaN        1                  1             NaN  

[5 rows x 357 columns]

In [30]:
english_names = OFSP_English.columns
rpt_rmv_names = OFSP_English_rpts_rmv.columns

# The names of the rpt_rmv columns are different making direct comparison challenging. 
# Will import the list of features used during machine learning classification

In [31]:
features = ['Sex', 'Age', 'Current_smoker', 'Former_smoker',
       'Sedentary', 'Height', 'Weight', 'Cervical_perimeter',
       'Abdominal_perimeter', 'Systolic_BP', 'Diastolic_BP',
       'Maxillofacial_profile', 'BMI', 'High_BP', 'Asthma', 'Rhinitis', 'COPD',
       'Respiratory_fail', 'Myocardial_infarct', 'Coronary_fail',
       'Arrhythmias', 'Stroke', 'Heart_fail', 'Arteriopathy', 'Gastric_reflux',
       'Glaucoma', 'Diabetes', 'Hypercholesterolemia', 'Hypertriglyceridemia',
       'Hypo(er)thyroidism', 'Depression', 'Obesity', 'Dysmorphology',
       'Restless_Leg_Syndrome', 'Snoring', 'Diurnal_somnolence',
       'Driving_drowsiness', 'Morning_fatigue', 'Morning_headache',
       'Memory_problem', 'Nocturnal_perspiration',
       'Shortness_of_breath_on_exertion', 'Nocturia', 'Drowsiness_accident',
       'Near_miss_accident', 'Respiratory_arrest', 'Epworth_scale',
       'Pichots_scale', 'Depression_scale']

In [32]:
all(item in rpt_rmv_names for item in features) # all features included in rpt_rmv df

True

In [33]:
all(item in english_names for item in features) # not all features included in OFSP_English df because they have different names

False

In [34]:
feat_set = set(features)
intersection = feat_set.intersection(english_names)
intersection # these features are in rpts_rmv and OFSP_English

{'Arrhythmias',
 'Arteriopathy',
 'Asthma',
 'COPD',
 'Depression',
 'Dysmorphology',
 'Glaucoma',
 'Hypertriglyceridemia',
 'Nocturia',
 'Obesity',
 'Rhinitis',
 'Sex',
 'Stroke'}

In [35]:
intersection = list(intersection) # converting to a list
type(intersection)

list

#### Checking to see if the data in the intersection columns is the same for all patients in OFSP_English and rpts_rmv

In [36]:
# removing the 1st row from OFSP_English for easier comparison. It contains French translation of column names
OFSP_English1 = OFSP_English.iloc[1: , :].reset_index(drop = True) 
OFSP_English1.head()

Patient ID Date of birth Department’s id Country’s ID  Sex Profession  \
0           1     21-Nov-73              33           74    2          5   
1           1           NaN             NaN          NaN  NaN        NaN   
2           1           NaN             NaN          NaN  NaN        NaN   
3           1           NaN             NaN          NaN  NaN        NaN   
4           2     10-Nov-33              17           74    1          7   

  Current smoker Number of current cigarettes package Former smoker   \
0              0                                  NaN              1   
1            NaN                                  NaN            NaN   
2            NaN                                  NaN            NaN   
3            NaN                                  NaN            NaN   
4              0                                  NaN              0   

  Number of former cigarettes package  ... Treatment name.38  \
0                                   5  ...               NaN   
1                                 NaN  ...               NaN   
2                                 NaN  ...               NaN   
3                                 NaN  ...               NaN   
4                                 NaN  ...               NaN   

  Treatment name.39 Short treatment name .18   ATC code.15 ATC class .19  \
0               NaN                      NaN           NaN           NaN   
1               NaN                      NaN           NaN           NaN   
2               NaN                      NaN           NaN           NaN   
3               NaN                      NaN           NaN           NaN   
4               NaN                      NaN           NaN           NaN   

  Dosage.32 Dosage.33 Table : Number of the visist Incorrect visit  
0       NaN       NaN       1                    1             NaN  
1       NaN       NaN       2                    2             NaN  
2       NaN       NaN       2                    3             NaN  
3       NaN       NaN       2                    4             NaN  
4       NaN       NaN       1                    1             NaN  

[5 rows x 390 columns]

In [37]:
print(OFSP_English1.shape)
print(OFSP_English_rpts_rmv.shape)

(190537, 390)
(190537, 357)


In [38]:
Intersect_English1 = OFSP_English1[intersection].apply(pd.to_numeric) 
Intersect_rpts_rmv = OFSP_English_rpts_rmv[intersection].apply(pd.to_numeric)

# this is to make sure all of the variables are being read as numeric so that when they are compared, no errors while arise
# new dfs with only the features common to both original dfs

In [39]:
Intersect_English1.equals(Intersect_rpts_rmv)

False

In [40]:
diff_rows = []
for i in range(190537):
    if Intersect_English1.iloc[i,:].equals(Intersect_rpts_rmv.iloc[i,:]):
        continue
    diff_rows.append(i)

# This creates a list of row indices in which the 2 dfs are different

In [41]:
len(diff_rows)

89919

In [43]:
Intersect_English1.iloc[1,:]

Asthma                  NaN
Obesity                 NaN
Dysmorphology           NaN
Arrhythmias             NaN
Glaucoma                NaN
Depression              NaN
Sex                     NaN
Stroke                  NaN
Arteriopathy            NaN
Nocturia                0.0
COPD                    NaN
Hypertriglyceridemia    NaN
Rhinitis                NaN
Name: 1, dtype: float64

In [44]:
Intersect_rpts_rmv.iloc[1,:]

Asthma                  0.0
Obesity                 1.0
Dysmorphology           NaN
Arrhythmias             0.0
Glaucoma                0.0
Depression              1.0
Sex                     NaN
Stroke                  0.0
Arteriopathy            0.0
Nocturia                0.0
COPD                    0.0
Hypertriglyceridemia    1.0
Rhinitis                0.0
Name: 1, dtype: float64

In [45]:
Hypertrig_cols = [col for col in english_names if 'Hypertrig' in col]
# Exploring why Hypertriglyceridemia may be different btwn dfs

In [46]:
Hypertrig_cols
# there are 2 hypertriglyceridemia columns

['Hypertriglyceridemia', 'Hypertriglyceridemia.1']

In [47]:
print(OFSP_English1.loc[1,'Hypertriglyceridemia'])
print(OFSP_English1.loc[1,'Hypertriglyceridemia.1'])

# So it looks like the rpts_rmv df only kept the column that had data

nan
1


### Looking deeper into which col names have been dropped

In [48]:
english_names_sorted = pd.Series(sorted(english_names))
rpt_rmv_names_sorted = pd.Series(sorted(rpt_rmv_names))
# going to export to excel because it's easier

#english_names_sorted.to_csv('english_names_sorted.csv')
#rpt_rmv_names_sorted.to_csv('rpt_rmv_names_sorted.csv')

##### Steps in Excel
1. Paste lists (from english_names_sorted and rpt_rmv_names_sorted) next to each other under appropriate headings
2. Columns name matches are placed next to each other. Ex: 'ATC class .1' vs 'ATCclass.1'
3. Columns in OFSP_English which have been removed from OFSP_rpts_rmv because they contain duplicate information
   have been inserted next to their duplicate name.
        OFSP_English     OFSP_rpts_rmv
    Ex1: 
        COPD             COPD
        COPD.1           COPD   
   Ex2: 
        Diabetes         Diabetes
         Diabetes       Diabetes
4. File saved as 'names_recreated.csv' 
5. Total duplicate names in OFSP_rpts_rmv column: 66  (So 33 names appeared twice)

This is the reason that OFSP_English contains 390 columns and OFSP_rpts_rmv contains 357 columns (390-33 = 357)
   

In [49]:
names_recreated = pd.read_csv("names_recreated.csv", encoding= 'unicode_escape', index_col = 0)

In [50]:
names_recreated.head()

OFSP_English                     OFSP_rpts_rmv
0         Shortness of breath on exertion   Shortness_of_breath_on_exertion
1   Shortness of breath on exertion scale  Shortnessofbreathonexertionscale
2                              __________                        __________
3                            __________.1                      __________.1
4                              ATC class                           ATCclass

In [51]:
rpt_features_eng = ['\xa0Patient ID','Arrhythmias','Arrhythmias.1','Arteriopathy','Arteriopathy.1','Asthma','Asthma.1','COPD','COPD.1',
                 'Coronary fail','Coronary fail.1','Current smoker','Current smoker.1','Depression','Depression.1',
                 '\xa0Diabetes ', 'Diabetes ','Dysmorphology','Dysmorphology.1','Former smoker ','Former smoker .1',
                 'Gastric reflux','Gastric reflux.1','Glaucoma','Glaucoma.1','Heart fail','Heart fail.1','High blood pressure',
                 'High blood pressure.1','Hypertriglyceridemia','Hypertriglyceridemia.1','Hypothyroidism / hyperthyroidism',
                 'Hypothyroidism / hyperthyroidism.1','Myocardial infarct','Myocardial infarct.1','Obesity','\xa0Obesity',
                 'Respiratory fail','Respiratory fail.1','Rhinitis','Rhinitis.1','Sedentary lifestyle','Sedentary lifestyle.1',
                 'Stroke','\xa0Stroke','\xa0Hypercholesterolemia', '\xa0Hypercholesterolemia.1']

# these are the repeated categories from OFSP_English that are also in the feature list *Patient ID also included for better tracking
# this will be used to determine which of the duplicates to keep

### Differences between data in columns with similar names in the OFSP_English df

In [52]:
pd.options.display.max_seq_items = 500
OFSP_English.columns

Index([' Patient ID', 'Date of birth', 'Department’s id', 'Country’s ID',
       'Sex', 'Profession', 'Current smoker',
       'Number of current cigarettes package', 'Former smoker ',
       'Number of former cigarettes package', 'Smoker ', 'Alcohol consumption',
       'Sedentary lifestyle', 'Asthma', 'Empty variable', 'Rhinitis', 'COPD',
       'Empty variable.1', 'Respiratory fail', 'ENT surgery', 'Other syrgery',
       'Allergy', ' ____', 'High blood pressure', 'Myocardial infarct',
       'Coronary fail', 'Arrhythmias', 'Stroke', 'Heart fail', 'Arteriopathy',
       'Gastric reflux', 'Glaucoma', ' Diabetes ', 'Type of diabetes',
       ' Hypercholesterolemia', 'Hypertriglyceridemia',
       'Hypothyroidism / hyperthyroidism', 'Depression',
       'Sleep apnea syndrome', 'Obesity', 'Dysmorphology', ' ____.1',
       ' User ID', 'Adel ID', 'Modification date', '_____', ' Visit ID',
       'Visit’s date', 'OPatient age at the time of visit', 'Current smoker.1',
       'Number of cu

In [53]:
OFSP_English[rpt_features_eng].head(20)

Patient ID      Arrhythmias  Arrhythmias.1     Arteriopathy  \
0   id_patient  troubleRythme_P  troubleRythme  arteriopathie_P   
1            1                0              0                0   
2            1              NaN              0              NaN   
3            1              NaN              0              NaN   
4            1              NaN              0              NaN   
5            2                0              0                0   
6            2              NaN              0              NaN   
7            3                0              0                0   
8            4                0              0                0   
9            5                0              0                0   
10           6                0            NaN                0   
11           7                0              0                0   
12           7              NaN              0              NaN   
13           7              NaN              0              NaN   
14           8                0              0                0   
15           9                0              0                0   
16          10                0              0                0   
17          11                0              0                0   
18          11              NaN              0              NaN   
19          12                0              0                0   

   Arteriopathy.1    Asthma Asthma.1    COPD COPD.1            Coronary fail  \
0   arteriopathie  asthme_P   asthme  bcpo_P   bcpo  insuffisanceCoronaire_P   
1               0         0        0       0      0                        0   
2               0       NaN        0     NaN      0                      NaN   
3               0       NaN        0     NaN      0                      NaN   
4               0       NaN        0     NaN      0                      NaN   
5               0         0        0       0      0                        0   
6               0       NaN        0     NaN      0                      NaN   
7               0         1        1       0      0                        0   
8               0         0        0       0      0                        0   
9               0         0        0       1      1                        1   
10            NaN         0      NaN       0    NaN                        0   
11              0         1        1       0      0                        0   
12              0       NaN        1     NaN      0                      NaN   
13              0       NaN        1     NaN      0                      NaN   
14              0         0        0       0      0                        0   
15              0         0        0       0      0                        0   
16              0         0        0       0      0                        0   
17              0         0        0       0      0                        0   
18              0       NaN        0     NaN      0                      NaN   
19              0         0        0       0      0                        0   

    ...    Respiratory fail Respiratory fail.1   Rhinitis Rhinitis.1  \
0   ...  insuffisanceResp_P   insuffisanceResp  rhinite_P    rhinite   
1   ...                   0                  0          0          0   
2   ...                 NaN                  0        NaN          0   
3   ...                 NaN                  0        NaN          0   
4   ...                 NaN                  0        NaN          0   
5   ...                   0                  0          0          0   
6   ...                 NaN                  0        NaN          0   
7   ...                   0                  0          1          1   
8   ...                   0                  0          0          0   
9   ...                   0                  0          0          0   
10  ...                   0                NaN          0        NaN   
11  ...                   0                  0     

#### Exploring which repeat column contains more information
These features are all binary, and it will be assumed that presence (1) is the more correct entry

Columns with higher sums (ie more presence of condition) will be kept

In [54]:
Eng_rpt_feat_df = OFSP_English[rpt_features_eng]
Eng_rpt_feat_df = Eng_rpt_feat_df.iloc[1: , 1:] # dropping 1st column (Patient IDs) and 1st row (French)
pd.options.display.max_columns = 500
Eng_rpt_feat_df.head(20)

Arrhythmias Arrhythmias.1 Arteriopathy Arteriopathy.1 Asthma Asthma.1 COPD  \
1            0             0            0              0      0        0    0   
2          NaN             0          NaN              0    NaN        0  NaN   
3          NaN             0          NaN              0    NaN        0  NaN   
4          NaN             0          NaN              0    NaN        0  NaN   
5            0             0            0              0      0        0    0   
6          NaN             0          NaN              0    NaN        0  NaN   
7            0             0            0              0      1        1    0   
8            0             0            0              0      0        0    0   
9            0             0            0              0      0        0    1   
10           0           NaN            0            NaN      0      NaN    0   
11           0             0            0              0      1        1    0   
12         NaN             0          NaN              0    NaN        1  NaN   
13         NaN             0          NaN              0    NaN        1  NaN   
14           0             0            0              0      0        0    0   
15           0             0            0              0      0        0    0   
16           0             0            0              0      0        0    0   
17           0             0            0              0      0        0    0   
18         NaN             0          NaN              0    NaN        0  NaN   
19           0             0            0              0      0        0    0   
20           0             0            0              0      0        0    0   

   COPD.1 Coronary fail Coronary fail.1 Current smoker Current smoker.1  \
1       0             0               0              0                0   
2       0           NaN               0            NaN                0   
3       0           NaN               0            NaN                0   
4       0           NaN               0            NaN                0   
5       0             0               0              0                0   
6       0           NaN               0            NaN                0   
7       0             0               0              0                0   
8       0             0               0              0                0   
9       1             1               1              0                0   
10    NaN             0             NaN              0              NaN   
11      0             0               0              1                1   
12      0           NaN               0            NaN                1   
13      0           NaN               0            NaN                1   
14      0             0               0              0                0   
15      0             0               0              0                0   
16      0             0               0              0                0   
17      0             0               0              0                0   
18      0           NaN               0            NaN                0   
19      0             0               0              0                0   
20      0             0               0              0                0   

   Depression Depression.1  Diabetes  Diabetes  Dysmorphology Dysmorphology.1  \
1           1            1          0         0           NaN             NaN   
2         NaN            1        NaN         0           NaN             NaN   
3         NaN            1        NaN         0           NaN             NaN   
4         NaN            1        NaN         0           NaN             NaN   
5           0            0          0         0             0               0   
6         NaN            0        NaN         0           NaN               0   
7           0            0          0         0             0               0   
8           1            1          0         0             0               0

In [55]:
type(Eng_rpt_feat_df.iloc[0, 0])

str

In [56]:
Eng_rpt_feat_df = Eng_rpt_feat_df.apply(pd.to_numeric) # converting to numeric
type(Eng_rpt_feat_df.iloc[0, 0])

numpy.float64

In [57]:
evens = []
for num in range(46):
    if num % 2 == 0:
        evens.append(num)
print(evens)

# indices over which for loop will be run

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44]


In [58]:
column_choice = []
for i in evens:
    if Eng_rpt_feat_df.iloc[: , i].sum() >= Eng_rpt_feat_df.iloc[: , (i+1)].sum():
        column_choice.append(Eng_rpt_feat_df.columns[i])
    column_choice.append(Eng_rpt_feat_df.columns[i+1])

In [59]:
column_choice

['Arrhythmias.1',
 'Arteriopathy.1',
 'Asthma.1',
 'COPD.1',
 'Coronary fail.1',
 'Current smoker.1',
 'Depression.1',
 'Diabetes ',
 'Dysmorphology.1',
 'Former smoker .1',
 'Gastric reflux.1',
 'Glaucoma.1',
 'Heart fail.1',
 'High blood pressure.1',
 'Hypertriglyceridemia.1',
 'Hypothyroidism / hyperthyroidism.1',
 'Myocardial infarct.1',
 '\xa0Obesity',
 'Respiratory fail.1',
 'Rhinitis.1',
 'Sedentary lifestyle.1',
 '\xa0Stroke',
 '\xa0Hypercholesterolemia.1']

### Converting the names in OFSP_English to the comparable names in OFSP_rpts_rmv

In [60]:
#subsetting OFSP_English by column_choice

OFSP_English2 = OFSP_English1[column_choice] # Here, I am using OFSP_English1 because it does not have the French row
OFSP_English2.head(20)

Arrhythmias.1 Arteriopathy.1 Asthma.1 COPD.1 Coronary fail.1  \
0              0              0        0      0               0   
1              0              0        0      0               0   
2              0              0        0      0               0   
3              0              0        0      0               0   
4              0              0        0      0               0   
5              0              0        0      0               0   
6              0              0        1      0               0   
7              0              0        0      0               0   
8              0              0        0      1               1   
9            NaN            NaN      NaN    NaN             NaN   
10             0              0        1      0               0   
11             0              0        1      0               0   
12             0              0        1      0               0   
13             0              0        0      0               0   
14             0              0        0      0               0   
15             0              0        0      0               0   
16             0              0        0      0               0   
17             0              0        0      0               0   
18             0              0        0      0               0   
19             0              0        0      0               0   

   Current smoker.1 Depression.1 Diabetes  Dysmorphology.1 Former smoker .1  \
0                 0            1         0             NaN                1   
1                 0            1         0             NaN                1   
2                 0            1         0             NaN                1   
3                 0            1         0             NaN                1   
4                 0            0         0               0                0   
5                 0            0         0               0                0   
6                 0            0         0               0                0   
7                 0            1         0               0                1   
8                 0            0         1               0                1   
9               NaN          NaN       NaN             NaN              NaN   
10                1            0         0               0                0   
11                1            0         0               0                0   
12                1            0         0               0                0   
13                0            0         0               0                0   
14                0            0         0               0                0   
15                0            0         0               0                0   
16                0            0         0               0                1   
17                0            0         0               0                1   
18                0            1         0               0                0   
19                0            0         0               0                0   

   Gastric reflux.1 Glaucoma.1 Heart fail.1 High blood pressure.1  \
0                 0          0            0                     1   
1                 0          0            0                     1   
2                 0          0            0                     1   
3                 0          0            0                     1   
4                 1          1            0                     1   
5                 1          1            0                     1   
6                 0          0            0                     0   
7                 1          0            0                     1   
8                 0          0            0                     1   
9               NaN        NaN          NaN                   NaN   
10                0          0            0                     1   
11                0          0            0                     1   
12                0          0        

In [61]:
column_choice_features = ['Arrhythmias', 'Arteriopathy', 'Asthma', 'COPD', 'Coronary_fail', 'Current_smoker', 'Depression', 
                          'Diabetes','Dysmorphology', 'Former_smoker', 'Gastric_reflux', 'Glaucoma', 'Heart_fail', 'High_BP',
                          'Hypertriglyceridemia','Hypo(er)thyroidism', 'Myocardial_infarct', 'Obesity', 'Respiratory_fail',
                          'Rhinitis', 'Sedentary', 'Stroke', 'Hypercholesterolemia']

In [62]:
print(len(column_choice))
print(len(column_choice_features))

23
23


In [63]:
#subsetting OFSP_English_rpts_rmv by column_choice_features

OFSP_English_rpts_rmv2 = OFSP_English_rpts_rmv[column_choice_features]
OFSP_English_rpts_rmv2.head(20)

Arrhythmias  Arteriopathy  Asthma  COPD  Coronary_fail  Current_smoker  \
0           0.0           0.0     0.0   0.0            0.0             0.0   
1           0.0           0.0     0.0   0.0            0.0             0.0   
2           0.0           0.0     0.0   0.0            0.0             0.0   
3           0.0           0.0     0.0   0.0            0.0             0.0   
4           0.0           0.0     0.0   0.0            0.0             0.0   
5           0.0           0.0     0.0   0.0            0.0             0.0   
6           0.0           0.0     1.0   0.0            0.0             0.0   
7           0.0           0.0     0.0   0.0            0.0             0.0   
8           0.0           0.0     0.0   1.0            1.0             0.0   
9           NaN           NaN     NaN   NaN            NaN             NaN   
10          0.0           0.0     1.0   0.0            0.0             1.0   
11          0.0           0.0     1.0   0.0            0.0             1.0   
12          0.0           0.0     1.0   0.0            0.0             1.0   
13          0.0           0.0     0.0   0.0            0.0             0.0   
14          0.0           0.0     0.0   0.0            0.0             0.0   
15          0.0           0.0     0.0   0.0            0.0             0.0   
16          0.0           0.0     0.0   0.0            0.0             0.0   
17          0.0           0.0     0.0   0.0            0.0             0.0   
18          0.0           0.0     0.0   0.0            0.0             0.0   
19          0.0           0.0     0.0   0.0            0.0             0.0   

    Depression  Diabetes  Dysmorphology  Former_smoker  Gastric_reflux  \
0          1.0       0.0            NaN            1.0             0.0   
1          1.0       0.0            NaN            1.0             0.0   
2          1.0       0.0            NaN            1.0             0.0   
3          1.0       0.0            NaN            1.0             0.0   
4          0.0       0.0            0.0            0.0             1.0   
5          0.0       0.0            0.0            0.0             1.0   
6          0.0       0.0            0.0            0.0             0.0   
7          1.0       0.0            0.0            1.0             1.0   
8          0.0       1.0            0.0            1.0             0.0   
9          NaN       NaN            NaN            NaN             NaN   
10         0.0       0.0            0.0            0.0             0.0   
11         0.0       0.0            0.0            0.0             0.0   
12         0.0       0.0            0.0            0.0             0.0   
13         0.0       0.0            0.0            0.0             0.0   
14         0.0       0.0            0.0            0.0             0.0   
15         0.0       0.0            0.0            0.0             0.0   
16         0.0       0.0            0.0            1.0             1.0   
17         0.0       0.0            0.0            1.0             1.0   
18         1.0       0.0            0.0            0.0             0.0   
19         0.0       0.0            0.0            0.0             0.0   

    Glaucoma  Heart_fail  High_BP  Hypertriglyceridemia  Hypo(er)thyroidism  \
0        0.0         0.0      1.0                   1.0                 0.0   
1        0.0         0.0      1.0                   1.0                 0.0   
2        0.0         0.0      1.0                   1.0                 0.0   
3        0.0         0.0      1.0                   1.0                 0.0   
4        1.0         0.0      1.0                   0.0                 0.0   
5        1.0         0.0      1.0                   0.0                 0.0   
6        0.0         0.0      0.0                   0.0                 0.0   
7        0.0         0.0      1.0                   0.0                 0.0   
8        0.0         0.0      1.0                   0.0                 0.0   
9        NaN        

In [64]:
print(type(OFSP_English2.iloc[0, 0]))
print(type(OFSP_English_rpts_rmv2.iloc[0, 0]))

<class 'str'>
<class 'numpy.float64'>


In [65]:
# converting OFSP_Englsih2 to numeric for comparison

OFSP_English2 = OFSP_English2.apply(pd.to_numeric)

print(type(OFSP_English2.iloc[0, 0]))
print(type(OFSP_English_rpts_rmv2.iloc[0, 0]))

<class 'numpy.float64'>
<class 'numpy.float64'>


In [66]:
OFSP_English2.head(20)

Arrhythmias.1  Arteriopathy.1  Asthma.1  COPD.1  Coronary fail.1  \
0             0.0             0.0       0.0     0.0              0.0   
1             0.0             0.0       0.0     0.0              0.0   
2             0.0             0.0       0.0     0.0              0.0   
3             0.0             0.0       0.0     0.0              0.0   
4             0.0             0.0       0.0     0.0              0.0   
5             0.0             0.0       0.0     0.0              0.0   
6             0.0             0.0       1.0     0.0              0.0   
7             0.0             0.0       0.0     0.0              0.0   
8             0.0             0.0       0.0     1.0              1.0   
9             NaN             NaN       NaN     NaN              NaN   
10            0.0             0.0       1.0     0.0              0.0   
11            0.0             0.0       1.0     0.0              0.0   
12            0.0             0.0       1.0     0.0              0.0   
13            0.0             0.0       0.0     0.0              0.0   
14            0.0             0.0       0.0     0.0              0.0   
15            0.0             0.0       0.0     0.0              0.0   
16            0.0             0.0       0.0     0.0              0.0   
17            0.0             0.0       0.0     0.0              0.0   
18            0.0             0.0       0.0     0.0              0.0   
19            0.0             0.0       0.0     0.0              0.0   

    Current smoker.1  Depression.1  Diabetes   Dysmorphology.1  \
0                0.0           1.0        0.0              NaN   
1                0.0           1.0        0.0              NaN   
2                0.0           1.0        0.0              NaN   
3                0.0           1.0        0.0              NaN   
4                0.0           0.0        0.0              0.0   
5                0.0           0.0        0.0              0.0   
6                0.0           0.0        0.0              0.0   
7                0.0           1.0        0.0              0.0   
8                0.0           0.0        1.0              0.0   
9                NaN           NaN        NaN              NaN   
10               1.0           0.0        0.0              0.0   
11               1.0           0.0        0.0              0.0   
12               1.0           0.0        0.0              0.0   
13               0.0           0.0        0.0              0.0   
14               0.0           0.0        0.0              0.0   
15               0.0           0.0        0.0              0.0   
16               0.0           0.0        0.0              0.0   
17               0.0           0.0        0.0              0.0   
18               0.0           1.0        0.0              0.0   
19               0.0           0.0        0.0              0.0   

    Former smoker .1  Gastric reflux.1  Glaucoma.1  Heart fail.1  \
0                1.0               0.0         0.0           0.0   
1                1.0               0.0         0.0           0.0   
2                1.0               0.0         0.0           0.0   
3                1.0               0.0         0.0           0.0   
4                0.0               1.0         1.0           0.0   
5                0.0               1.0         1.0           0.0   
6                0.0               0.0         0.0           0.0   
7                1.0               1.0         0.0           0.0   
8                1.0               0.0         0.0           0.0   
9                NaN               NaN         NaN           NaN   
10               0.0               0.0         0.0           0.0   
11               0.0               0.0         0.0           0.0   
12               0.0               0.0         0.0           0.0   
13               0.0               0.0         0.0           0.0   
14               0.0               0.0         0.0           0.0   
15              

In [67]:
OFSP_English2.equals(OFSP_English_rpts_rmv2)

False

In [68]:
diff_rows2 = []
for i in range(190537):
    if OFSP_English2.iloc[i,:].equals(OFSP_English_rpts_rmv2.iloc[i,:]):
        continue
    diff_rows2.append(i)

# This creates a list of row indices in which the 2 dfs are different

In [69]:
len(diff_rows2)

190537

### Not sure why this is happening